### PR TITLE
Replace random art tab with critique tab

### PR DESCRIPTION
**Pull request by charmander**
Because when have you ever wanted to see 11 random submissions out of the last 256?

- corrects searching within browse categories
- allows browsing critique requests in chronological order
- splits index cache
- redesigns stream and site update sections based on weasyl2014
- fixes comment hiding on site updates
- makes narrow viewport design use home tabs instead of expandable sections (I fixed section closing and state indication first, but then this was better)
- makes thumbnail layout algorithm do what was originally intended
- removes unnecessarily large padding from thumbnail grids when JavaScript hasn’t run [yet]
- reduces recent submissions on homepage to medium footprint (two rows at max width) and removes their heading to place tabs and part of recently popular above the fold on many common viewport sizes
- limits homepage critique selection to one per person
- allows critique thumbnails to use consistently-sized and WebP thumbnails (plus any future changes)
- probably more

Sorry this is so entangled.

### CODE REVIEW

**Review by kfkitsune — CHANGES_REQUESTED** [review]
One thing that needs to be fixed, one thing that might not have been introduced here (and just a byproduct of the dev DB being a partial export), and one being an amusing observation of GitHub deciding to have a negative-one and zeroth lines in a diff? (See Gitter/Matrix for the screencapture.)

File: weasyl/controllers/general.py [hunk 1]
```diff
@@ -1,21 +1,37 @@
...
+        vary=["Cookie"],  # SFW mode, sign in/out, account changes
+    )
+
+
+_BROWSE = {
+    'submit': ("Submissions", "Browse Submissions"),
+    'char': ("Characters", "Browse Characters"),
+    'journal': ("Journals", "Browse Journals"),
+    'critique': ("Critique requests", "Browse Critique Requests"),
+}
```
[the remark this answers is not in the document]
Comment (charmander) [reply weasyl/controllers/general.py:34]: Browse Users isn’t a thing (current https://www.weasyl.com/search?find=user is submissions), but yes, this should be fixed. Thanks!

File: weasyl/templates/admincontrol/siteupdate.html [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
     <label for="siteupdatecontent">Content</label>
-    <textarea class="markdown input expanding last-input" name="content" rows="9" id="siteupdatecontent" required>${update.content}</textarea>
+    <textarea class="markdown input expanding last-input" name="content" rows="9" id="siteupdatecontent" required>${update['content']}</textarea>
     <br />
 
     <h3>Options</h3>
 
     <label class="input-checkbox">
-      <input type="checkbox" name="wesley"$:{' checked' if update.wesley else ''}>
+      <input type="checkbox" name="wesley"$:{' checked' if update['wesley'] else ''}>
```
[the remark this answers is not in the document]
Comment (charmander) [reply weasyl/templates/admincontrol/siteupdate.html:21]: Yep, that’s exactly what it is. Pick your favorite existing user id, edit `config/weasyl-staff.example.py`, run `./wzl configure`.

```python
wesley = 34165
```

**Review by kfkitsune — APPROVED** [review]
As far as I can tell, it looks good to me!

**Comment by charmander** [comment]
Thank you for the review!

**Pull request merged**